pkg/source/imgpkg.go

```diff
@@ -52,16 +69,29 @@ func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, image s
 	return fmt.Sprintf("%s@%s", uploadRef.Name(), digestRef.DigestStr()), nil
 }
 
-type ggcrRemoteOptionsStashKey struct{}
+type registryOptionsStashKey struct{}
+type containerRemoteTransportStashKey struct{}
+
+func StashRegistryOptions(ctx context.Context, options registry.Opts) context.Context {
+	return context.WithValue(ctx, registryOptionsStashKey{}, options)
+}
+
+func RetrieveRegistryOptions(ctx context.Context) *registry.Opts {
+	options, ok := ctx.Value(registryOptionsStashKey{}).(registry.Opts)
+	if !ok {
+		return nil
+	}
+	return &options
+}
 
-func StashGgcrRemoteOptions(ctx context.Context, options ...remote.Option) context.Context {
-	return context.WithValue(ctx, ggcrRemoteOptionsStashKey{}, options)
+func StashContainerRemoteTransport(ctx context.Context, rTripper http.RoundTripper) context.Context {
+	return context.WithValue(ctx, containerRemoteTransportStashKey{}, rTripper)
 }
 
-func RetrieveGgcrRemoteOptions(ctx context.Context) []remote.Option {
-	options, ok := ctx.Value(ggcrRemoteOptionsStashKey{}).([]remote.Option)
+func RetrieveStashContainerRemoteTransport(ctx context.Context) *http.RoundTripper {
```

Comment:
the next three suggestions are part of a single change, but GitHub won't let me offer the suggestion in one go
```suggestion
func RetrieveStashContainerRemoteTransport(ctx context.Context) http.RoundTripper {
```